src/unix/solarish/mod.rs

```diff
@@ -206,6 +217,24 @@ s! {
         pub ai_next: *mut addrinfo,
     }
 
+    pub struct shmid_ds {
+        pub shm_perm: ipc_perm,
+        pub shm_segsz: ::size_t,
+        pub shm_flags: ::uintptr_t,
```

Comment:
There is no shm_flags in illumos.  This is also at the location where shm_amp would be for illumos. In addition, shm_gransize and shm_allocated are not defined on illumos, and the padding appears to be incorrect for illumos.

Reply:
https://github.com/illumos/illumos-gate/blob/master/usr/src/uts/common/sys/shm.h#L88-L115 shows the illumos definition (as @bgermann said, currently there is no 32-bit support for rust on illumos, so it's probably not necessary to add the 32-bit versions -- if that were to ever happen, it'd probably better be done as its own change since many more things would need to be updated).

Reply:
The IPC_\* values are ok -- for likely historical reasons lost to time, the values are defined in octal on illumos (but appear to match what you have)

Reply:
Oh, yes, so forget about that.